backend/src/main/kotlin/io/zell/zdb/log/LogContent.kt

```diff
@@ -56,4 +54,39 @@ class LogContent {
     override fun toString(): String {
         return "{ \"records\": [${records.joinToString()}] } "
     }
+
+    fun asDotFile(): String {
+        val content = StringBuilder("digraph log {\n")
+
+        for (record in records) {
+            if (record is ApplicationRecord) {
+                for (entry in record.entries) {
```

Comment:
🔧 This is quite some nesting and might benefit from a functional style collection pipeline, e.g. records.stream.filter.etc

Reply:
Good point :) Often easier to me to just write down the loops :laughing: